boa/src/exec/identifier/mod.rs

```diff
@@ -1,21 +1,17 @@
 use super::{Executable, Interpreter};
-use crate::{
-    builtins::value::{ResultValue, Value, ValueData},
-    syntax::ast::node::identifier::Identifier,
-};
+use crate::{builtins::value::ResultValue, syntax::ast::node::identifier::Identifier};
 
 impl Executable for Identifier {
     fn run(&self, interpreter: &mut Interpreter) -> ResultValue {
-        let reference = resolve_binding(interpreter, self.as_ref());
-        match reference.data() {
-            ValueData::Undefined => Err(interpreter
+        let reference = interpreter
+            .realm()
+            .environment
+            .get_binding_value(self.as_ref());
+        match reference {
+            Some(value) => Ok(value),
+            None => Err(interpreter
                 .throw_reference_error(self.as_ref())
                 .expect_err("throw_reference_error() must return an error")),
```

Comment:
It's clearer if we use `.construct_reference_error()`, which return `Value` sow we can return `Err(interpreter.construct_reference_error(msg))`

Reply:
I think we should use `ok_or_else()` here, to make it more readable, with the function you mention as the parameter.

Reply:
Applied